Terminal.Gui/Core/Graphs/Annotations.cs

```diff
@@ -0,0 +1,325 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
```

Comment:
I have a feeling having this project depend on `System.Drawing` is a bad idea:

![image](https://user-images.githubusercontent.com/585482/115570146-baddc680-a272-11eb-9bb2-2068e6a436da.png)

from https://docs.microsoft.com/en-us/dotnet/api/system.drawing?view=net-5.0

I'm pretty sure the classes you are actually using (e.g. `RectangleF`) are fine, but there's a risk.

I think we should ask @migueldeicaza  for an opinion on which option is best:

0) Add the `System.Drawing.Primitives` nuget package, which includes all of these primitives. It continues to use the same namespace. https://www.nuget.org/packages/System.Drawing.Primitives/4.3.0?_src=template
1) Just use `System.Drawing` and continue to be careful not to pull in classes that actually have a dependency on GDI+.
2) Fork the classes we want from `System.Drawing` into `/Terminal.Gui/Types`
3) Take a dependency on another floating point library with similar functions.

My reading indicates option 0 is the best, but I don't fully understand what DLLs end up getting used in the end and if this actually provides an alternative to using the full `System.Drawing` assemblies.

I read through this https://github.com/dotnet/runtime/issues/21980 where @migueldeicaza participated and the answer is still not clear to me.

Reply:
These are good concerns and challenges.

What we have done is copy the types from System.Drawing into the code (see Terminal.Gui/Types/Size.cs., Rectangle, Point for example).   So we should do the same in this case.

Reply:
Thanks for the guidance.

I have copied the code from https://github.com/dotnet/corefx/tree/master/src/System.Drawing.Primitives/src/System/Drawing into the Types dir.  

I made the following changes

- Changed namespace to `Terminal.Gui` 
- Updated cref in comments to refer to `Terminal.Gui` version
- Removed `readonly` keyword from the private properties (which required csharp 8)
- Implement GetHashCode (I just copied the definition from Rectangle/Point/Size - adding ).

Reply:
Excellent!